addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/handler/YamahaBridgeHandler.java

```diff
@@ -128,49 +147,59 @@ private void setupRefreshTimer(int initialWaitTime) {
         if (refreshTimer != null) {
             refreshTimer.cancel(false);
         }
-        refreshTimer = scheduler.scheduleWithFixedDelay(new Runnable() {
-            @Override
-            public void run() {
-                updateAllZoneInformation();
-            }
-        }, initialWaitTime, refrehInterval, TimeUnit.SECONDS);
+        refreshTimer = scheduler.scheduleWithFixedDelay(() -> updateAllZoneInformation(), initialWaitTime,
+                refrehInterval, TimeUnit.SECONDS);
     }
 
     /**
-     * Periodically and initially called. Updates
+     * Periodically and initially called. This must run in another thread, because
+     * all update calls are blocking.
      */
     void updateAllZoneInformation() {
+        logger.trace("updateAllZoneInformation");
         try {
-            SystemControl basicDeviceInformation = new SystemControl();
-            basicDeviceInformation.fetchDeviceInformation(xml, state);
-            basicDeviceInformation.fetchPowerInformation(xml, state);
-
-            updateProperty(YamahaReceiverBindingConstants.PROPERTY_VERSION, state.version);
-            updateProperty(YamahaReceiverBindingConstants.PROPERTY_ASSIGNED_NAME, state.name);
-
+            DeviceInformation deviceInformation = ProtocolFactory.DeviceInformation(connection, deviceInformationState);
+            deviceInformation.update();
+            zoneDiscoveryService.publishZones(deviceInformationState, thing.getUID());
+
+            SystemControl systemControl = ProtocolFactory.SystemControl(connection, this);
+            systemControlState.power = true;
```

Comment:
Actually I need to, but I forgot to document it here. It would result in an endless loop if I don't.
@zarusz: Can you add a comment above that line with:

> Set power = true before calling systemControl.update(), otherwise the systemControlStateChanged method would call updateAllZoneInformation() again